### DISCUSSION TITLE
relationship name

### DISCUSSION THREAD

**Discussion by soederpop** [supabase/supabase#5907] [Questions] [answered]
I have a schema such as:

- Tenant
   - Has Many Portfolios
   - Has Many Projects
   
- Portfolio
  - Belongs to Tenant
  - Has Many Projects

- Project
  - Belongs to Tenant
  - Belongs to Portfolio
  
Is it true that it uses the name of the constraint as defined in the SQL?

I can't do the following because constraints have to be uniquely named

```sql
CREATE TABLE public.projects (
    ....
   CONSTRAINT tenant FOREIGN KEY (tenant_id) REFERENCES public.tenants(id)
);

CREATE TABLE public.portfolios (
    ....
   CONSTRAINT tenant FOREIGN KEY (tenant_id) REFERENCES public.tenants(id)
);
```

But this means I can't do

```js
await supa.from('portfolios').select('*, tenant(id)')
await supa.from('projects').select('*, tenant(id)')
```

I have to do 

```js
await supa.from('portfolios').select('*, portfolio_tenant(id)')
await supa.from('projects').select('*, project_tenant(id)')
```

which is just ugly.  

how can i control the name of the relationship?

**Comment by GaryAustin1** [comment]
Did you try your first example (assuming 'tenant' is the table name, nothing to do with the constraint name)?
Normally I just use the name of the table (with a foreign key pointing to it) or the name of the foreign key column.  Either seems to work.
https://supabase.com/docs/reference/javascript/select

**Comment by steve-chavez** [comment, marked as answer]
> I can't do the following because constraints have to be uniquely named

Actually that's not true, the constraint names are namespaced by table. So you can have two constraints with the same name.

**Reply by soederpop** [reply]
you're right.  I was getting an error in the migrations, but I must not have read it closely enough.  I can re-use constraint names and that seems to make everything work as expected!
